src/interp.rs

```diff
@@ -353,7 +363,23 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
                 } else {
                     panic!("{:?}", msg);
                 }
-            }
+            },
+            TerminatorKind::SwitchInt {
+                ref discr,
+                ref switch_ty,
+                ref values,
+                ref targets
+            } => {
+                let d = self.eval_operand(discr).val.to_u128();
+                let mut idx = values.len();
```

Comment:
I think this can be simplified to something like:

```rust
let idx = values.iter().find(|x| *x == d).unwrap_or_else(|| values.len());
```

Reply:
Simplified to something similar but using enumeration instead as it's the index I am after